### PR TITLE
Moving method from pulpcore-plugin models

### PR DESCRIPTION
**Pull request by fao89**
Moving all the methods for:
Content, ContentGuard, Remote and Publisher
from pulpcore-plugin to pulpcore

ref #5353
https://pulp.plan.io/issues/5353

Required PR: https://github.com/pulp/pulpcore/pull/335

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/plugin/download/__init__.py [hunk 1]
```diff
@@ -1,4 +1,9 @@
-from .base import BaseDownloader, DownloadResult  # noqa
-from .factory import DownloaderFactory  # noqa
-from .file import FileDownloader  # noqa
-from .http import http_giveup, HttpDownloader  # noqa
+
+from pulpcore.download import (  # noqa
```
Comment (bmbouter) [inline pulpcore/plugin/download/__init__.py]: I didn't expect the downloaders to move. Did they have to move?

Comment (fabricio-aguiar) [reply pulpcore/plugin/download/__init__.py]: Due to `Remote`: Remote needs a DownloaderFactory: https://github.com/pulp/pulpcore-plugin/blob/master/pulpcore/plugin/models/remote.py#L46
I mentioned about it here:
https://pulp.plan.io/issues/5353#note-11

**Comment by bmbouter** [comment]
@fabricio-aguiar this needs conflict resolution

**Review by bmbouter — APPROVED** [review]
Thanks @fabricio-aguiar! :+1:

**Pull request merged**